### DISCUSSION TITLE
When change page I need to refresh

### DISCUSSION THREAD

**Discussion by albertopallotto** [rappasoft/laravel-livewire-tables#1053] [Q&A] [answered]
Hi,
I use rappasoft for all tables of my project.
I use the mount function to pass parameters on table class, and I use these parameters on builder function.
When I change page of table result, the parameters is null, and I see the wrong results, if I refresh the page all is correct.
How can I fix it?

**Comment by lrljoe** [comment, marked as answer]
> Hi,
> I use rappasoft for all tables of my project.
> I use the mount function to pass parameters on table class, and I use these parameters on builder function.
> When I change page of table result, the parameters is null, and I see the wrong results, if I refresh the page all is correct.
> How can I fix it?

Are you using v1 or v2?

If you declare the variables as public, and set them when you include the component, does it work as expected?

Does it update when you sort the page (for example)?

**Reply by albertopallotto** [reply]
I use v2.
I had mistakenly set the variable as protected.
thanks for the solution

**Comment by milanchheda** [comment]
I am facing the exact problem even after setting the variable as public

```html
<livewire:user-table :users="$users" />
```

And then in the component
```php
public $users;
```

```php
Column::make('Name', 'authenticatable_id')
                ->sortable()
                ->format(function ($value) {
                    return isset($this->users[$value]) ? $this->users[$value] : '-';
                }),
```

This returns the name properly first time but then shows `-` for every pagination. What am I missing? 🤔

**Reply by milanchheda** [reply]
Nevermind. I had mistakenly set the pageName and variable name same due to which it was getting override.
